### ISSUE TITLE
[gdal:x64-windows-static] build failure

### ISSUE THREAD

**Issue by stronger1234** [microsoft/vcpkg#7375]
**Host Environment**
- OS: [Windows 10]
- Compiler: MSVC 2019(v142)

**To Reproduce**
`./vcpkg install osg:x64-windows-static`

**Failure logs** 
```
Starting package 1/10: gdal:x64-windows-static
Building package gdal[core]:x64-windows-static...
-- Using cached C:/vcpkg/downloads/gdal241.zip
-- Extracting source C:/vcpkg/downloads/gdal241.zip
-- Applying patch C:/vcpkg/ports/gdal/0001-Fix-debug-crt-flags.patch
-- Applying patch C:/vcpkg/ports/gdal/0002-Fix-static-build.patch
-- Applying patch C:/vcpkg/ports/gdal/0003-Fix-std-fabs.patch
-- Extracting source C:/vcpkg/downloads/gdal241.zip
-- Applying patch C:/vcpkg/ports/gdal/0001-Fix-debug-crt-flags.patch
-- Applying patch C:/vcpkg/ports/gdal/0002-Fix-static-build.patch
-- Applying patch C:/vcpkg/ports/gdal/0003-Fix-std-fabs.patch
-- Building x64-windows-static-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:58 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.21.27702/bin/Hostx64/x64/nmake.exe" -f makefile.vc GDAL_HOME=C:\vcpkg\packages\gdal_x64-windows-static DATADIR=C:\vcpkg\packages\gdal_x64-windows-static\share\gdal HTMLDIR=C:\vcpkg\packages\gdal_x64-windows-static\share\gdal\html GEOS_DIR=C:\vcpkg\installed\x64-windows-static\include "GEOS_CFLAGS=-IC:\vcpkg\installed\x64-windows-static\include -DHAVE_GEOS" PROJ_INCLUDE=-IC:\vcpkg\installed\x64-windows-static\include EXPAT_DIR=C:\vcpkg\installed\x64-windows-static\include EXPAT_INCLUDE=-IC:\vcpkg\installed\x64-windows-static\include CURL_INC=-IC:\vcpkg\installed\x64-windows-static\include "SQLITE_INC=-IC:\vcpkg\installed\x64-windows-static\include " PG_INC_DIR=C:\vcpkg\installed\x64-windows-static\include OPENJPEG_ENABLED=YES OPENJPEG_CFLAGS=-IC:\vcpkg\installed\x64-windows-static\include OPENJPEG_VERSION=20100 WEBP_ENABLED=YES WEBP_CFLAGS=-IC:\vcpkg\installed\x64-windows-static\include LIBXML2_INC=-IC:\vcpkg\installed\x64-windows-static\include PNG_EXTERNAL_LIB=1 PNGDIR=C:\vcpkg\installed\x64-windows-static\include ZLIB_INC=-IC:\vcpkg\installed\x64-windows-static\include ZLIB_EXTERNAL_LIB=1 MSVC_VER=1900 WIN64=YES PROJ_FLAGS=-DPROJ_STATIC CURL_CFLAGS=-DCURL_STATICLIB CXX_CRT_FLAGS=/MT PROJ_LIBRARY=C:\vcpkg\installed\x64-windows-static\lib\proj.lib PNG_LIB=C:\vcpkg\installed\x64-windows-static\lib\libpng16.lib GEOS_LIB=C:\vcpkg\installed\x64-windows-static\lib\libgeos.lib EXPAT_LIB=C:\vcpkg\installed\x64-windows-static\lib\expat.lib "CURL_LIB=C:\vcpkg\installed\x64-windows-static\lib\libcurl.lib wsock32.lib wldap32.lib winmm.lib" "SQLITE_LIB=C:\vcpkg\installed\x64-windows-static\lib\sqlite3.lib " OPENJPEG_LIB=C:\vcpkg\installed\x64-windows-static\lib\openjp2.lib WEBP_LIBS=C:\vcpkg\installed\x64-windows-static\lib\webp.lib LIBXML2_LIB=C:\vcpkg\installed\x64-windows-static\lib\libxml2.lib ZLIB_LIB=C:\vcpkg\installed\x64-windows-static\lib\zlib.lib
    Working Directory: C:/vcpkg/buildtrees/gdal/src-x64-windows-static-release/gdal-2.4.1
    Error code: 2
    See logs for more information:
      C:\vcpkg\buildtrees\gdal\nmake-build-x64-windows-static-release-out.log
      C:\vcpkg\buildtrees\gdal\nmake-build-x64-windows-static-release-err.log

Call Stack (most recent call first):
  ports/gdal/portfile.cmake:252 (vcpkg_execute_required_process)
  scripts/ports.cmake:74 (include)


Error: Building package gdal:x64-windows-static failed with: BUILD_FAILED
```
[nmake-build-x64-windows-static-release-err.log](https://github.com/microsoft/vcpkg/files/3419759/nmake-build-x64-windows-static-release-err.log)
[nmake-build-x64-windows-static-release-out.log](https://github.com/microsoft/vcpkg/files/3419760/nmake-build-x64-windows-static-release-out.log)

**Additional context**
I was confirmed using the latest portfiles with `git pull`, `.\vcpkg update` and `.\vcpkg upgrade --no-dry-run`.
Additionally, I try the non-static build and it succeeded.

**Comment by JackBoosY** [comment]
Hi @stronger1234, thanks for reporting this issue.
I've fixed this issue, please update vcpkg and rebuild gdal.

Thanks.

**Comment by ChisonShang** [comment]
Excuse me, I also have this problem when I use “vcpkg.exe install osg:x64-windows” to install OSG. But my vcpkg is the latest version and this issue was not be fixed.

**Comment by JackBoosY** [comment]
@ChisonShang  Please open an new issue, and don't forgot to attach your failure logs.

Thanks.

**Comment by ChisonShang** [comment]
> @ChisonShang Please open an new issue, and don't forgot to attach your failure logs.
> 
> Thanks.

I have located where my problem is.
In the log, there are several  warnings have the same part
 "cl : Command line warning D9027 : source file '**Files**\vcpkg\vcpkg-master\installed\x64-windows\include' ignored"
But my path vcpkg.exe is "**...:\Program Files\...**", maybe the space in my path was considered separator. So when I moved vcpkg.exe to another location without space in the path, this problem disappeared.
